polars/polars-arrow/src/kernels/rolling/no_nulls/variance.rs

```diff
@@ -57,20 +81,20 @@ impl<'a, T: NativeType + IsFloat + std::iter::Sum + AddAssign + SubAssign + Mul<
 
         // we traverse all values and compute
         if T::is_float() && recompute_sum {
-            self.sum_of_squares = self
+            self.sum_of_powers = self
                 .slice
                 .get_unchecked(start..end)
                 .iter()
-                .map(|v| *v * *v)
+                .map(|v| v.pow(self.base))
                 .sum::<T>();
         } else {
             for idx in self.last_end..end {
                 let entering_value = *self.slice.get_unchecked(idx);
-                self.sum_of_squares += entering_value * entering_value;
```

Comment:
I think a more generic `SumOfPower` is welcome, but I think we should keep the `SumSquaredWindow` as is. 
I've gone through a great effort to make the kernels efficient and doing a more generic power instead of a simple multiply will revert some of that efficiency.